docs/Dialects/Sim/RationaleSim.md

```diff
@@ -0,0 +1,22 @@
+# Simulation (Sim) Dialect Rationale
+
+This document describes various design points of the `sim` dialect, why it is
+the way it is, and current status. This follows in the spirit of other [MLIR
+Rationale docs](https://mlir.llvm.org/docs/Rationale/).
+
+## Introduction
+
+The `sim` dialect provides a high-level representation for simulator-specific
+operations. The purpose of the dialect is to provide a high-level representation
+for constructs which interact with simulators (Verilator, VCS, Arc, ...) that
```

Comment:
This is going to be hard overall.  Attributes/pragmas interact with the simulators, do those go here?  What is the layering w.r.t. sv or (hypothetically) vhdl or arc?